apps/cli-daemon/src/app/workspace-manager/workspace-manager.controller.ts

```diff
@@ -0,0 +1,27 @@
+import { Directory } from '@angular-cli-gui/shared/data';
+import {
+  Controller,
+  Get,
+  InternalServerErrorException,
+  Query,
+} from '@nestjs/common';
+
+import { WorkspaceManagerService } from './workspace-manager.service';
+
+@Controller('workspace-manager')
+export class WorkspaceManagerController {
+  constructor(
+    private readonly workspaceManagerService: WorkspaceManagerService
+  ) {}
+
+  @Get('dir?:path')
+  public async getDirectory(@Query('path') path: string): Promise<Directory[]> {
+    try {
+      return path
+        ? this.workspaceManagerService.getDirectory(path)
+        : this.workspaceManagerService.getRootDirectory();
```

Comment:
not sure about this logic, maybe should be in a separate API and leave this one clean (without fallback to root) ?
@agroupp @OmerGronich WDYT ?